### ISSUE TITLE
Make browser auto-reloading a bit more discerning? (feature request)

### ISSUE THREAD

**Issue by ddanielou** [bdkjones/CodeKit-1-Old#57]
Hoping the issues's not been raised already. Would it be desirable for CodeKit to keep the browser from refreshing a tab that's got nothing to do with the saved file, à la LiveReload/Xrefresh? Perhaps by setting the project's url, or by reloading http://localhost only?

Just wanted to check about another bug I was writing about in this very textarea and lost what I typed because of the refresh. Bad.

**Comment by bdkjones** [comment]
Agreed.

The next build will have the option to reload either:

1) The active tab/window in each browser
2) All tabs/windows with URLs that contain "localhost"

The code is complete and I've tested it -- it works. On a related note: I hate writing AppleScript.

**Comment by ddanielou** [comment]
Fantastic.

But... (hrm, sorry).
Well, my local server is set up with a bunch of VirtualHosts that don't necessarily contain "localhost". Damn. Not going to work. I don't know if:
a) my way of doing things is the way of doing things,
b) it would be hard to parse the /etc/hosts file for anything that maps to 127.0.0.1 (probably a bit overkill ?),
c) it wouldn't be easier really to simply tell CodeKit which url must be reloaded.

Anyway, thanks a lot for your work!

**Comment by bdkjones** [comment]
Well, I can't touch the hosts file. That would be a great way to be instantly rejected from the App Store. 

Can you give me more information on your workflow? Why are you using virtual hosts, and is there any pattern to them that could be used to hook the refresh? 

I don't want to require users to specify a URL for each project... the refresh feature should "just work" without any setup. I could conceivably add an app-wide preference to let the user specify a string to search for in the URL, but that's a bit of a kludge. 

This current approach will cover 95% of users, I'd imagine.

**Comment by ddanielou** [comment]
For each of my sites there's a virtual host setup like so:

DocumentRoot "/Users/***/Sites/example.com"
ServerName local.example.com

And in /etc/hosts:

127.0.0.1    local.example.com

I do it mainly to be able to use absolute paths for images, css and the like when developing; /img wouldn't do if the url was, say, http://localhost/website.com.

It wouldn't be too hard to match local.*, but it wouldn't be very reliable either. I bet there's other websites out there that use this word too.

I could rename my virtual host to example.com.localhost or something if it helped CodeKit, but I don't really feel like doing that.

If such a workflow is very uncommon, then I guess I can live without this feature anyway.

Then again, if you want it to "just work", there might be a way to get this information from the developer's IDE? "Local URL" in Coda, probably something along these lines in other apps?

**Comment by bdkjones** [comment]
Any reason you're not using MAMP? If you're not familiar with it, it's a complete Apache/PHP stack. You point it to a folder on your drive and that folder becomes the root of the web server.

So if you point it to: ~/Desktop/myWebsite/

Then you open your browser and go to:

http://localhost/

The page that comes up is ~/Desktop/myWebsite/index.html and you can use root-relative links in your files with no issues. Plus, you don't have to go edit the hosts file and MAMP runs completely separately from the Apache/PHP stack in OS X itself. 

It's clean, handy and fast.

**Comment by ddanielou** [comment]
> Any reason you're not using MAMP?

I use XAMPP. Honestly, I don't know why. Because the version with all the nice features is paid, perhaps? I can't remember exactly, I just know that for some reason, some time ago, I chose XAMPP over MAMP, and that, as far as I know, XAMPP doesn't offer that feature – at least not with a nice GUI.

Still, I like being able to work on many websites "at once" without having to point localhost to a different folder all the time. It's not as much about microseconds of work shaved here and there as about keeping things separated, having urls that are close to what they'll be in production, … Oh, and it's also about having those nice preview images in Coda's Site list, too.… !

Again, not that big of a deal, I'm not even sure I'd use the feature anyway.
